### PR TITLE
Implement series API query hint to not hit chunks

### PR DESCRIPTION
**Pull request by gouthamve**
Fixes https://github.com/prometheus/prometheus/issues/5547

Inspired from https://github.com/prometheus/prometheus/issues/5547#issuecomment-620032812

I've confirmed the series API works, and it works via remote-read too.

Signed-off-by: Goutham Veeramachaneni <gouthamve@gmail.com>

### CODE REVIEW

File: tsdb/querier.go [hunk 1]
```diff
@@ -120,18 +120,23 @@ func NewBlockQuerier(b BlockReader, mint, maxt int64) (storage.Querier, error) {
...
 		return storage.ErrSeriesSet(err)
 	}
 	if sortSeries {
 		p = q.index.SortedPostings(p)
 	}
+
+	if hints != nil {
+		mint = hints.Start
+		maxt = hints.End
+		if hints.Func == "series" {
```
[the remark this answers is not in the document]
Comment (gouthamve) [reply tsdb/querier.go:134]: Yep, I thought about changing it to `metadata`, but given we will persist and query HELP and TYPE info in the future, I think `series` is better.

**Comment by brian-brazil** [comment]
:+1:

**Review by codesome — APPROVED** [review]

File: tsdb/querier.go (hunk 1 above)
[the remark this answers is not in the document]
Comment (bwplotka) [reply tsdb/querier.go:134]: hah we had similar discussion with @yeya24 recently. We called `series, labels, labelval` apis just `labels` in our place (:

File: web/api/v1/api.go [hunk 2]
```diff
@@ -602,10 +602,16 @@ func (api *API) series(r *http.Request) (result apiFuncResult) {
 		q.Close()
 	}
 
+	hints := &storage.SelectHints{
+		Start: timestamp.FromTime(start),
+		End:   timestamp.FromTime(end),
+		Func:  "series", // There is no series function, this token is used for lookups that don't need samples.
```
Comment (bwplotka) [inline web/api/v1/api.go:608]: Would be nice to have constant IMO, referenced to tsdb, no?

**Review by bwplotka — COMMENTED** [review]
Some post-merge comment. Thanks for this!

**Pull request merged**